src/Silex/Provider/SecurityServiceProvider.php

```diff
@@ -570,6 +571,14 @@ public function register(Container $app)
 
             $app['validator.validator_service_ids'] = array_merge($app['validator.validator_service_ids'], array('security.validator.user_password' => 'security.validator.user_password_validator'));
         }
+
+        if (class_exists('Symfony\Bundle\SecurityBundle\ArgumentResolver\UserArgumentResolver')) {
```

Comment:
and you also need to check for the existence of the interface in HttpKernel first, to be sure that you don't use a recent SecurityBundle version with an older HttpKernel.